### DISCUSSION TITLE
Global and per-profile exclude list

### DISCUSSION THREAD

**Discussion by m3nu** [borgbase/vorta#854] [Ideas]
Just an idea. Would be interested in whether this is feasible.

**Is your feature request related to a problem? Please describe.**
Most OSs have a number of default locations that can be excluded from backups. Like logs and cache files. Creating these exclusion lists is quite some work and is repeated by each user.

**Describe the solution you'd like**
I suggest to have managed default exclude lists for each OS that can be included with a checkbox in addition to user-provided exclusions. Example of some exclusions that are likely the same for everyone:

```
**/.DS_Store
**/.tox
**/.vagrant
**/__pycache__
~/.Trash
~/.android/cache
~/.android/build-cache
```

**Comment by ThomasWaldmann** [comment]
I don't think this is a good idea, just because of the many expected different cases.

Just for the samples you have given:

- **/.DS_Store - macOS specific, usually not seen on pure linux or windows
- **/.tox - likely only found on python developer machines
- **/.vagrant - likely only on developer machines
- **/__pycache__ - likely only found on python developer machines
- ~/.Trash - likely only on specific(?) desktop machines
- ~/.android/cache - likely only on android developer machines
- ~/.android/build-cache - likely only on android developer machines

Instead of adding some comfort (less to type), it might also confuse users ("what's this vagrant thing?").

**Comment by m3nu** [comment]
Thanks for weighing in. You're right. The actual list of files is probably too specific. Github tried to make such lists and they are [very comprehensive](https://github.com/github/gitignore/tree/master/Global). 

How about a global user-defined list that can be applied to a single profile with a checkbox. Roughly like this:

- *Misc* tab becomes *Global Settings*
- *Global Settings* has a section with *Global Exclusions*
- Every repo has a checkbox above exclusions *Use Global Exclusions*

This would solve the problem of retyping the same exclusions. Since no one else seems to have this problem, let's make it low-prio until there were some more comments.

**Comment by ThomasWaldmann** [comment]
If you manage exclusions (or more generally: include/exclude patterns), order of them might be problematic. Maybe not a problem for simple stuff, but definitely if it gets more complex.

Also I am not sure if just dealing with exclusions would be best for vorta or whether it should have the more recent / more general `--patterns-*` mechanism.

**Comment by aritorhamo** [comment]
Missing collection of default exclusions was the first thing I noticed when I arrived to the sources tab. Every other back-up software that I have used has had one. It would be very handy, especially because creating exclusions is so complicated in Vorta (I in fact realized that Vorta is not for me when I saw how complicated it is). Here's the user guide for creating exclusions, which opens by clicking the context link on the Sources tab:  
[borg help patterns](https://borgbackup.readthedocs.io/en/stable/usage/help.html#borg-help-patterns)
It seems to be written by a computer scientist for computer scientists. Unfortunately I'm not one :-).

**Comment by atfrase** [comment]
Since OSX comes with a standard default backup tool already (Time Machine), and therefore other mac software presumably knows how to add its own backup-unnecessary paths to those exclusions, what about at least adding a checkbox for mac users to just "exclude everything that Time Machine would exclude by default"?

According to [this article](https://www.howtogeek.com/294600/save-space-on-your-time-machine-drive-by-excluding-these-folders-from-backups/), those exclusions seem to be stored in `/System/Library/CoreServices/backupd.bundle/Contents/Resources/StdExclusions.plist` and/or accessible via `sudo mdfind "com_apple_backup_excludeItem = 'com.apple.backupd'"`.

**Comment by m3nu** [comment]
Interesting. That gives a long list with some useful items, like Chrome favicons. I wonder how this is edited and whether this list makes sense for users.
